### PR TITLE
Feature/handle lists of enums in enum conversion

### PR DESCRIPTION
**Pull request by michael-gut-ergon**
Most of the files are clean up - PR should have been splitted...
- jcenter() deprecated
- kotlin upgrade 1.3.72 -> 1.5.30
- code style

The real action is in 
 - TypeUtil.kt 
 - CblFieldHolder

### CODE REVIEW

File: couchbase-entity/src/main/java/com/kaufland/model/mapper/MapifyHolder.kt [hunk 1]
```diff
@@ -1,23 +1,13 @@
...
 import com.kaufland.model.mapper.type.MapifyElementType
-import com.kaufland.util.FieldExtractionUtil
-import com.squareup.kotlinpoet.TypeName
-import com.squareup.kotlinpoet.asTypeName
-import kaufland.com.coachbasebinderapi.mapify.Mapify
 import kaufland.com.coachbasebinderapi.mapify.Mapifyable
 import kaufland.com.coachbasebinderapi.mapify.Mapper
 import java.io.Serializable
-import java.math.BigDecimal
+import java.util.*
```
Comment (flgubler-ergon) [inline couchbase-entity/src/main/java/com/kaufland/model/mapper/MapifyHolder.kt:9]: don't we have ktlint on entity-framework?

File: couchbase-entity/src/main/java/com/kaufland/model/field/CblFieldHolder.kt [hunk 2]
```diff
@@ -132,7 +132,11 @@ class CblFieldHolder(field: Field, allWrappers: List<String>) : CblBaseFieldHold
 
     private fun evaluateClazzForTypeConversion(): TypeName {
         return if (isIterable) {
-            TypeUtil.string()
+            if (TypeUtil.isMap(fieldType)) {
```
Comment (flgubler-ergon) [inline couchbase-entity/src/main/java/com/kaufland/model/field/CblFieldHolder.kt:135]: this is the main change of this PR, isn't it? (the one relevant for supporting lists of ENUMs)

**Review by flgubler-ergon — APPROVED** [review]
Can't say that I understood everything but I did not see anything bad :-)

File: couchbase-entity/src/main/java/com/kaufland/model/mapper/MapifyHolder.kt (hunk 1 above)
Comment (sbra0902) [reply couchbase-entity/src/main/java/com/kaufland/model/mapper/MapifyHolder.kt:9]: poorly not yet but I think we should add it soon 💯

**Review by sbra0902 — APPROVED** [review]

**Pull request merged**